### DISCUSSION TITLE
Non-zero thickness grain boundaries

### DISCUSSION THREAD

**Discussion by kngavl** [neperfepx/neper#192] [Q&A] [answered]
Is it possible to create a polycrystal but with non-zero grain boundary thickness? I would like to make a core-shell like structure to model grain boundary diffusion in a polycrystal.

I saw in a previous post that it was impossible to refine the mesh around grain boundaries but would it be possible to generate a simple polycrystal and then "shrink" the individual crystal faces towards their voronoi centers? 

By using: neper -T -n 100 -dim 2 -transform "scale(0.2,0.2)", I can generate thick boundaries around the crystals but I'm not sure how to treat them after that i.e. can those boundaries be deleted or meshed.

**Comment by Sofia-Y** [comment]
Hi, @kngavl!

I am actually working on a similar topic( diffusion in polycrystalline materials). I want to double-check with you on how to give a thickness to the boundaries around each crystal. I tried the command you have listed above but it did not work for me. Do you mind sharing a picture of your tessellation (no meshing just the tesselation)? Maybe explain the command you have used. I couldn't find an example in the manual.  

As for your actual question, I am not sure what software you are using for your analysis. I would suggest you get the tessellation from Neper and use your finite element software to control your meshing. 

Thanks.

**Reply by kngavl** [reply]
Hi Sofia, the above command worked fine for me as is but I don't actually know what is fundamentally produced.
If you look in the documentation:
https://neper.info/docs/neper.pdf
You can search for the transformation command in chapter 2 (P. 20).

Unfortunately I've uninstalled Neper and I can't find any of the pictures I generated as I've formulated my work in a different direction to get around this tricky problem

**Comment by rquey** [comment]
Neper does not directly enable for the shrinkage of grains and insertion of non zero boundaries.

The command that you provided scales the whole tessellation and does not generate any kind of thick boundaries.

Scaling the cells toward their centers would not be difficult to do, but it does not generate boundaries of the same width everywhere. You can get an idea of how it would work as follows (`explode` moves cells away from the tess center, which is not exactly what you want):
```
$ neper -T -n 100 -dim 2
$ neper -M n100-id1.tess -transform "explode(0.2)" -interface cohesive
```
(Open the resulting mesh with Gmsh).

**Reply by kngavl** [reply]
Thank you for your reply and for taking the time to explain what the code actually does.

It would be really great if Neper could do this in the future ... ;)

**Reply by dharmabas** [reply]
Hi guys, has this problem been revisited? I am looking for a similar polycrystal model as mentioned above.

**Reply by rquey** [reply]
No, it hasn't...

**Reply by bseligmann** [reply]
Dear @rquey, 
you mention that 'scaling cells toward their centers would not be difficult'. Could you comment on how to do that, as I am intending to surround grains with a seam of another phase. This seam is irregular in itself, so the same width issue would not concern me. 

Best regards and thank you.

**Reply by rquey** [reply, marked as answer]
@bseligmann - Give a try to these two commands:
```
neper -T -n 100 -dim 2
neper -M n100-id1.tess -transform "explode(0.2)" -interface cohesive
```
and open `n100-id1.msh` with Gmsh.

You'll see that what it would essentially take is to 1/ shrink the grains (which is not exactly what `explode(0.2)` does) and 2/ create interface elements (which is almost what `-interface cohesive` does, only triangle elements around the original vertices are missing).

**Comment by GreenBird12** [comment]
Is neper currently can generate core-shell structure,such as shrink every grain cell
![Snipaste_2025-03-22_22-26-11](https://github.com/user-attachments/assets/d462b42a-9d7f-479b-9a46-41de38245692)

**Reply by rquey** [reply]
No.
